src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/basm/BasmService.kt

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.model.basm
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.welcometoprison.model.MoveType
+import uk.gov.justice.digital.hmpps.welcometoprison.model.Movement
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Includes
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Location
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.People
+import uk.gov.justice.digital.hmpps.welcometoprison.model.basm.Model.Profile
+import java.time.LocalDate
+import kotlin.reflect.KClass
+
+@Service
+class BasmService(private val basmClient: BasmClient) {
+
+  fun <T : Includes> Map<String, Map<String, Includes>>.getAs(key: String, type: KClass<T>): (id: String) -> T? {
```

Comment:
The includes section is a list of unrelated types

This data structure is after that get's converted into a nested map structure of `entity type` -> `id` -> `entity` 

This extension function provides type safety access over that. 

provide type safe access to includes section of json api response.

Th